internal/driver/settings.go

```diff
@@ -60,6 +56,17 @@ func writeSettings(fname string, settings *settings) error {
 	if err != nil {
 		return fmt.Errorf("could not encode settings: %w", err)
 	}
+
+	// create the settings directory if it does not exist
+	fdir := filepath.Dir(fname)
+	if _, err := os.Stat(fdir); os.IsNotExist(err) {
```

Comment:
Why is this check needed? I.e. why not os.MkdirAll(fdir, 0700)?

Reply:
Great suggestion. Mostly because I didn't think of it. It turns out: this stat is exactly what MkdirAll does. Oops! Removing this now.